common/overlaysynccleanuphandler.cpp

```diff
@@ -32,3 +32,8 @@ ReturnCode OverlaySyncCleanupHandler::clean()
 {
     return RecursiveCopy::getInstance().copy(m_src, m_dst);
 }
+
+const std::string OverlaySyncCleanupHandler::queryName()
+{
+    return "";
```

Comment:
This does not make sense. I would say that the destination path would be what we want to check here? The reason for that would be that it is valid to mount the same location to many places (in theory), but not to mount several locations to the same place.